app/shared/cms/halremote.ts

```diff
@@ -0,0 +1,98 @@
+import {Http, Headers, RequestOptionsArgs} from 'angular2/http';
+import {Observable} from 'rxjs';
+import TemplateParser from 'url-template';
+import {HalRemoteCache} from './halremote.cache';
+
+export interface HalLink {
+  title: string ;
+  name: string;
+  profile: string;
+  href: string;
+  templated: boolean;
+}
+
+/**
+ * Http layer for HAL requests
+ */
+export class HalRemote {
+
+  constructor(
+    private http: Http,
+    private host: string,
+    private token: string
+  ) {}
+
+  expand(link: HalLink, params: {} = null): string {
+    if (!link || !link.href) {
+      return null;
+    } else if (link.templated) {
+      return TemplateParser.parse(this.host + link.href).expand(params || {});
+    } else {
+      return this.host + link.href;
+    }
+  }
+
+  get(link: HalLink, params: {} = null): Observable<{}> {
+    let href = this.expand(link, params);
+    if (href) {
+      let cachedResponse = HalRemoteCache.get(href);
+      if (cachedResponse) {
+        return cachedResponse;
+      } else {
+        return HalRemoteCache.set(href, this.http.get(href, this.httpOptions()).flatMap((res) => {
+          if (res.status === 200) {
+            return Observable.of(res.json());
+          } else {
+            return Observable.throw(new Error(`Got ${res.status} from GET ${href}`));
+          }
+        }));
```

Comment:
But then again, looks like you aren't doing this often, so perhaps fine to just be in get, and keep the cache simple.